docker/dockerfile.merlin

```diff
@@ -88,17 +100,26 @@ RUN ln -s /usr/bin/python3 /usr/bin/python
 # https://gitlab.kitware.com/cmake/cmake/-/issues/24119
 # A fix has already been merged but not yet released:
 # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7859
-# 2023-02-22: pynvml==11.5.0 is currently incompatible with our version of dask/distributed
-RUN pip install --no-cache-dir --upgrade pip; pip install --no-cache-dir "cmake<3.25.0" ninja scikit-build pandas==1.3.5 \
-                cupy-cuda117 nvidia-pyindex pybind11 pytest \
+RUN pip install --no-cache-dir --upgrade pip; pip install --no-cache-dir "cmake<3.25.0" ninja scikit-build pandas==1.3.5
+
+# cupy-cuda wheels come from a different URL on aarch64
+RUN if [ $(uname -m) == "aarch64" ] ; then \
+        pip install cupy-cuda115 -f https://pip.cupy.dev/aarch64/ ; \
```

Comment:
This logic implies that we are working with cuda 11. I think that in march we are moving to cuda 12. Are we going to need to support cuda 11.x for ARM support when that shift happens?

Reply:
I think that's a discussion for a different PR.  This PR only adds arm64 support to the existing Dockerfile.  If we need to change the CUDA version, that's a different PR and completely independent of adding arm64 support.  Both CUDA 11 and CUDA 12 work fine on arm64, so we could go either way, but arm64 should match x86 regardless.